python/ml4ir/base/model/architectures/dnn.py

```diff
@@ -60,13 +67,21 @@ def get_op(layer_type, layer_args):
         ]
 
     def get_architecture_op(self):
-        def _architecture_op(ranking_features):
+        def _architecture_op(ranking_features, metadata_features):
             layer_input = ranking_features
 
             # Pass ranking features through all the layers of the DNN
             for layer_op in self.layer_ops:
                 layer_input = layer_op(layer_input)
 
+            if 'positional_bias_handler' in self.model_config and \
+                    self.model_config['positional_bias_handler']['key'] == \
+                    PositionalBiasHandler.FIXED_ADDITIVE_POSITIONAL_BIAS:
+
+                positional_bias = self.positional_bias_layer(metadata_features[self.feature_config.get_rank()['name']],
+                                                             max_ranks=self.model_config['positional_bias_handler']['max_ranks_count'], training=True)
+                layer_input = layers.Add()([positional_bias, layer_input])
+
```

Comment:
@mohazahran Why do we need this if we are adding it to DNN layer?

Reply:
Or rather, why do we need the one in the constructor? Just create the object here.